### DISCUSSION TITLE
Welcome to silabs-firmware-builder Discussions!

### DISCUSSION THREAD

**Discussion by Nerivec** [Nerivec/silabs-firmware-builder#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by snippem** [comment]
I think my question is more applicable here.

@Nerivec a question maybe there is a abvious solution (i hope so)
I own several silabs sticks and want to use my dongle-e sticks as routers. Now i have 2 network as production 1 running Z-stack and one Ember. On the Z-stack network the dongle-e router works with your software. But when i want the dongle-e router on the ember network it does not get recognized. In my opinion this is deu to the fact both mac-adresses are the same on the router i have : 0xe0798dfffea7bd73
and the coordinator has the same adres: 0xe0798dfffea7bd73. Is there a way to change the mac-adres of the routers? As the second dongle-e is not recognized on the Z-stack network maybe deu to the fact that the mac-adress is the same to.

**Reply by Nerivec** [reply]
You have 3 sticks with the same EUI? That's unfortunate...
At the moment, you can only change the EUI for a coordinator firmware.

**Reply by snippem** [reply]
> You have 3 sticks with the same EUI? That's unfortunate... At the moment, you can only change the EUI for a coordinator firmware.

Okay that is not an option now because I am of course using my coordinator.
Or it should be possible to change the mac address without negative consequences on my production network.
I am interested in how do you change the mac address and is this permanent? Maybe I can change the address by first flashing the coordinator version and then flashing the router version.

**Reply by Nerivec** [reply]
Silabs uses the NVM3 to provide an alternative EUI (the original is kept, but overridden if the one in the NVM3 is set).
Using a coordinator firmware, followed by router, might work because the NVM3 is kept between flashings, but I've never tried it.
https://github.com/Nerivec/ember-zli/wiki/Stack#write-eui64-nvm3-token

**Reply by snippem** [reply]
> Silabs uses the NVM3 to provide an alternative EUI (the original is kept, but overridden if the one in the NVM3 is set). Using a coordinator firmware, followed by router, might work because the NVM3 is kept between flashings, but I've never tried it. https://github.com/Nerivec/ember-zli/wiki/Stack#write-eui64-nvm3-token


I tried it and changed 1 value of the mac address.
Unfortunately when you flash the router firmware the original mac comes back. Maybe other possibilities?

update:
Tried different flash programs but this does not make a difference. (ember-cli and universal-silabs-flasher).

**Reply by Nerivec** [reply]
I guess the router firmware code isn't triggering the logic to override the EUI. If that's the case, there isn't much we will be able to do.
I suppose you could find someone near you (not too near though 😁) to exchange adapters with, so you can "remove" the problem entirely.
Only immediately feasible option I see is to change the EUI of your coordinator instead, but you would have to re-create that network from scratch and re-pair all devices.

**Reply by MattWestb** [reply]
EFR32 chips is having the original MAC burned in the factors and cant being changed.
Standard EZSP have implanting used data that can being set in the factory and / or with commands and it have one reserved place for one override the factory MAC.
Then its in the user data flash it can being written ones and then not being changed from the application.
If having one SWD flasher is possible reading the user-data and deleting the customer MAC in the bin file and then erase the flash page and writing back the updated user-data and its using the factory MAC and can being updated ones agen from the application.
If having one newer bootloader it shall also being possible patching the user-data but its one very deep operation and i have not  seen any manufacture doing that also after shipping devices with false / corrupt data settings (IKEA some lights and some remotes with wrong device ID strings).
